app/test/square_size_test.go

```diff
@@ -174,28 +191,50 @@ func (s *SquareSizeIntegrationTest) SetupBlockSizeParams(t *testing.T, squareSiz
 
 	require.NoError(t, s.cctx.WaitForNextBlock())
 
-	// query the proposal to get the id
-	govQueryClient := govv1.NewQueryClient(s.cctx.GRPCClient)
-	propResp, err := govQueryClient.Proposals(s.cctx.GoContext(), &govv1.QueryProposalsRequest{ProposalStatus: govv1.StatusVotingPeriod})
-	require.NoError(t, err)
-	require.Len(t, propResp.Proposals, 1)
+	proposalID := uint64(0)
 
-	// create and submit a new msgVote
-	msgVote := govv1.NewMsgVote(testfactory.GetAddress(s.cctx.Keyring, testnode.DefaultValidatorAccountName), propResp.Proposals[0].Id, govv1.OptionYes, "")
-	res, err = txClient.SubmitTx(s.cctx.GoContext(), []sdk.Msg{msgVote}, blobfactory.DefaultTxOpts()...)
-	require.NoError(t, err)
-	require.Equal(t, abci.CodeTypeOK, res.Code)
+	// try to query a few times until the voting period has passed
+	for i := 0; i < 20; i++ {
+		// query the proposal to get the id
+		govQueryClient := govv1.NewQueryClient(s.cctx.GRPCClient)
+		propResp, err := govQueryClient.Proposals(s.cctx.GoContext(), &govv1.QueryProposalsRequest{ProposalStatus: govv1.StatusVotingPeriod})
+		require.NoError(t, err)
```

Comment:
noticed this queries were flakey, so trying to fix that by querying a bunch instead of trying to time it once